go/host/rpc/clientapi/client_api_filter.go

```diff
@@ -38,6 +38,15 @@ func (api *FilterAPI) Logs(ctx context.Context, encryptedParams common.Encrypted
 		return nil, fmt.Errorf("could not subscribe for logs. Cause: %w", err)
 	}
 
+	// We return the ID of the newly-created subscription, before returning any log events.
```

Comment:
Since there's no way for us to get the subscription ID based on the returned subscription object, we send a first packet that's just the subscription ID (and no log) instead.